api/v1alpha2/zz_generated.conversion.go

```diff
@@ -607,6 +613,7 @@ func autoConvert_v1alpha3_AWSMachineSpec_To_v1alpha2_AWSMachineSpec(in *v1alpha3
 	out.IAMInstanceProfile = in.IAMInstanceProfile
 	out.PublicIP = (*bool)(unsafe.Pointer(in.PublicIP))
 	out.AdditionalSecurityGroups = *(*[]AWSResourceReference)(unsafe.Pointer(&in.AdditionalSecurityGroups))
+	// WARNING: in.FailureDomain requires manual conversion: does not exist in peer-type
```

Comment:
How should downconversion be handled due to this? Should we just fail downconversion if FailureDomain is set? Or should we override AvailabilityZone with the value from FailureDomain?

Reply:
That's a good point, lemme check

Reply:
Added the manual conversion from FD -> AZ, if AZ is empty